### PR TITLE
feat(metrics): add findings by product breakdown

### PR DESCRIPTION
**Pull request by amchiclet**
Requires interface change: https://github.com/semgrep/semgrep-interfaces/pull/210

Closes PA-3312.

Breakdown findings by product. The product name in the metrics matches the product name in the CLI options.

Tested by running semgrep CI. Breakdown by products show up in Metabase.

Also updated a snapshot of a test to support this new field..

### CODE REVIEW

File: cli/src/semgrep/metrics.py
Comment (emjin) [inline cli/src/semgrep/metrics.py]: This seems pretty fundamental, maybe we should put it in `constants.py` instead?

File: cli/src/semgrep/metrics.py
Comment (emjin) [inline cli/src/semgrep/metrics.py:314]: Unrelated to your PR but what is "kept"?

**Review by emjin — APPROVED** [review]
LGTM. I'm assuming you tested with some runs that used all three products

File: cli/src/semgrep/metrics.py
Comment (amchiclet) [reply cli/src/semgrep/metrics.py]: Done.

File: cli/src/semgrep/metrics.py
Comment (amchiclet) [reply cli/src/semgrep/metrics.py:314]: "Kept" here is the opposite of "ignored". So it stores whatever matches that weren't ignored. If I understand correctly, when [the user uses `#nosemgrep`](https://github.com/semgrep/semgrep/blob/986e7772c6dfe1f174e919ad8fa0d6f4749a4c85/cli/src/semgrep/nosemgrep.py#L4-L5), that match will be ignored.

**Comment by amchiclet** [comment]
Yep, tested with code, supply-chain, and secrets enabled. Waiting for CI, then will merge if it passes.

**Pull request merged**
